### PR TITLE
fix: ZCAP middleware panic

### PR DESCRIPTION
**Pull request by llorllale**
The aries framework and json-gold libraries are adding documents to the `CachingDocumentLoader` while processing json-ld documents. This risks concurrent read/write of `CachingDocumentLoader`, which is backed by a regular map.

We work around this by creating a new `CachingDocumentLoader` for every verification. Documents are loaded once.

Signed-off-by: George Aristy <george.aristy@securekey.com>

### CODE REVIEW

File: pkg/auth/zcapld/service.go [hunk 1]
```diff
@@ -48,13 +44,10 @@ func New(keyManager kms.KeyManager, crypto cryptoapi.Crypto, storeProv ariesstor
 		return nil, fmt.Errorf("failed to open store %s: %w", storeName, err)
 	}
 
-	ldDocumentLoader, err := createJSONLDDocumentLoader()
-	if err != nil {
-		return nil, fmt.Errorf("failed create json ld document loader: %w", err)
-	}
+	// TODO re-enable cached jsonld document loader: https://github.com/trustbloc/edv/issues/173.
```
Comment (fqutishat) [inline pkg/auth/zcapld/service.go:47]: this performance killer

**Review by aholovko — APPROVED** [review]

**Review by fqutishat — CHANGES_REQUESTED** [review]

File: pkg/auth/zcapld/service.go (hunk 1 above)
Comment (llorllale) [reply pkg/auth/zcapld/service.go:47]: Hi @fqutishat 

Turns out the framework is adding a cache anyway (see #173 ).

Comment (llorllale) [reply pkg/auth/zcapld/service.go:47]: [This is where it's added](https://github.com/hyperledger/aries-framework-go/blob/bf8af862aa8358f2a91cb0d2856bd366919f795b/pkg/doc/signature/jsonld/processor.go#L264-L272).

The cache map is passed in [here](https://github.com/hyperledger/aries-framework-go/blob/bf8af862aa8358f2a91cb0d2856bd366919f795b/pkg/doc/signature/proof/jws.go#L123-L122).

[These](https://github.com/hyperledger/aries-framework-go/blob/bf8af862aa8358f2a91cb0d2856bd366919f795b/pkg/doc/signature/proof/cache.go#L9-L13) are the cached JSON-LD contexts of the framework. Notice it has both context we need (https://w3id.org/security/v1, https://w3id.org/security/v2).

Comment (llorllale) [reply pkg/auth/zcapld/service.go]: @fqutishat please check now - the JSONLD documents are cached, but a new `CachingDocumentLoader` is used for every verification.

**Review by fqutishat — APPROVED** [review]

**Pull request merged**